compiler/main/PhaseTracker.cpp

```diff
@@ -27,117 +27,115 @@
 #include <cstring>
 #include <algorithm>
 
-struct SortByTime
-{
-  bool operator() (Pass const& a, Pass const& b) const
-  {
+struct SortByTime {
+  bool operator()(Pass const& a, Pass const& b) const {
     return a.CompareByTime(b);
   }
 };
+struct SortByMemory {
+  bool operator()(Pass const& a, Pass const& b) const {
+    return a.CompareByMemory(b);
+  }
+};
 
-static void PassesSortByTime(std::vector<Pass>& passes);
-
-static void PassesReport(const std::vector<Pass>& passes,
-                         unsigned long            totalTime);
+static void PassesReport(const std::vector<Pass>& passes, PhaseData total);
 
-static void PassesReport(FILE*                    fp,
-                         const std::vector<Pass>& passes,
-                         unsigned long            totalTime);
+static void
+PassesReport(FILE* fp, const std::vector<Pass>& passes, PhaseData total);
 
 /************************************* | **************************************
-*                                                                             *
-* Implementation of PhaseTracker                                              *
-*                                                                             *
-************************************** | *************************************/
+ *                                                                             *
+ * Implementation of PhaseTracker                                              *
+ *                                                                             *
+ ************************************** | *************************************/
```

Comment:
If intentional... consider removing the space after the first `*` on each line to keep the right side of the box aligned.

Reply:
I have reverted this